syntax = "proto3";
import "common_proto/empty.proto";
import "common_proto/tip.proto";
import "common_proto/common.proto";
import "component_proto/scene_comp.proto";
option cc_generic_services = true;

message StartCrossGsRequest
{
  IpPort rpc_client = 1;
  IpPort rpc_server = 2;
  uint32 gs_node_id = 3;
  uint32 server_type = 4;
}

message StartCrossGsResponse
{
  repeated SceneInfo scenes_info = 1;
}

message StartControllerRequest
{
  IpPort rpc_client = 1;
  IpPort rpc_server = 2;
  uint32 controller_node_id = 3;
}

message EnterCrossMainSceneRequest
{
  uint64 scene_id = 1;
  uint64 player_id = 2;
}

message EnterCrossMainSceneResponese
{
  Tips error = 1;
  uint64 scene_id = 2;
  uint64 player_id = 3;
}

message EnterCrossRoomSceneSceneResponse
{
  Tips error = 1;
}

message EnterCrossMainSceneWeightRoundRobinRequest
{
  uint32 scene_confid = 1;
  uint64 player_id = 2;
}

message EnterCrossRoomSceneSceneWeightRoundRobinResponse
{
  Tips error = 1;
}

message LeaveCrossMainSceneRequest
{
  uint64 player_id = 1;
}

message GameConnectToControllerRequest
{
}

service LobbyService 
{
  rpc StartCrossGs (StartCrossGsRequest) returns (StartCrossGsResponse);
  rpc StartControllerNode (StartControllerRequest) returns (Empty);
  rpc EnterCrossMainScene (EnterCrossMainSceneRequest) returns (EnterCrossMainSceneResponese);
  rpc EnterCrossMainSceneWeightRoundRobin (EnterCrossMainSceneWeightRoundRobinRequest) returns (EnterCrossRoomSceneSceneWeightRoundRobinResponse);
  rpc LeaveCrossMainScene (LeaveCrossMainSceneRequest) returns (Empty);
  rpc GameConnectToController (GameConnectToControllerRequest) returns (Empty);
}


